src/components/container/OverviewTable/OverviewTable.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { CSSProperties, useContext, useState } from 'react';
+import { ColorDefinition, computeThresholdDays, getDailyDataTypeDefinition, getSurveyDataProvider, isSurveyDataType, NotEnteredThreshold, OverviewData, OverviewDataProvider, OverviewDataType, PrimaryOverviewDataType, resolveColor, SecondaryOverviewDataType, useInitializeView } from '../../../helpers';
+import { LayoutContext, LoadingIndicator } from '../../presentational';
+import './OverviewTable.css';
+import { add, startOfToday } from 'date-fns';
+import { createPreviewDataProvider } from './OverviewTable.previewData';
+import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
+import { FontAwesomeSvgIcon } from 'react-fontawesome-svg-icon';
+import language from '../../../helpers/language';
+
+export interface OverviewTableProps {
+    preview?: boolean;
+    primaryDataType: PrimaryOverviewDataType;
+    secondaryDataTypes: SecondaryOverviewDataType[];
+    daysToCompute?: number;
+    primaryHeaderBackgroundColor?: ColorDefinition;
+    primaryHeaderTextColor?: ColorDefinition;
+    goodValueBackgroundColor?: ColorDefinition;
+    goodValueTextColor?: ColorDefinition;
+    goodValueIndicator?: IconDefinition;
+    notGoodValueBackgroundColor?: ColorDefinition;
+    notGoodValueTextColor?: ColorDefinition;
```

Comment:
Both great questions.

> 1. Why _must_ a value be either good or not good?  I realize you can set "not good" to be a neutral color, and then your chart is only highlighting "good" and "normal" - but it feels like most of the time these sorts of thresholds are ternary instead of binary?  If I get under 6 hours of sleep it's not good, if I get 6-7 it's normal, and if I get over 7 it's good?  That sort of situation?

The original/existing reason for these colors was/is to be able to highlight and provide positive feedback when values are in some "good" range.  For my use case, that means having a minimum number of steps (i.e. 6k+).  For sleep, it means having enough sleep, but not too much (i.e. 7-10 hours).  Thus far, this has always been used in a binary fashion.  Things are either "good" and are highlighted as such, or they are put into the other bucket (currently referred to as "not good").  That isn't to say that those other values are necessarily "bad".  It just means they are not being highlighted as "good".  I'm certainly open to naming suggestions for the "values which are not considered good" category.  Maybe just "other"? 🙂

> 2. Similarly, it feels like we're avoiding using the term "bad" here - notably that there's not a "notGoodValueIndicator."  Are we really avoiding letting someone highlight "problem" areas with this graph?

The original use case for this widget was as part of a mental health study.  I think to some degree, there was purpose in not focusing on the negatives, but rather, just highlighting the positives and allowing the negatives (or non-positives) to be inferred.  That said, I wouldn't be opposed to adding support for negative highlighting as well, in case some future use case calls for that.

The `goodValueIndicator` is new and has not been used previously.  It is my attempt at making this table a tad more "accessible" for the color averse folks in the population.  Regardless of the color palette chosen, it facilitates marking the "positive" values with a non-color based indication.  The ones without the indication are the "non-positives".  Since the categorization is currently binary (i.e. good, other), I didn't think a second indicator was necessary.

I think if we do add support for negative highlighting, which creates a third categorization (i.e. good, bad, other), then I do think it would be appropriate to add a `badValueIndicator` option as well.

Reply:
I have renamed "notGood" to "other" for now.  I will also see about adding support for highlighting "bad" values.